docker/Dockerfile

```diff
@@ -41,34 +38,30 @@ ENV PIP_USE_PEP517=1
 FROM python-base AS pyproject-builder
 
 # Install dependencies
-RUN \
-  --mount=type=cache,target=/var/cache/apt,sharing=locked \
-  --mount=type=cache,target=/var/lib/apt,sharing=locked \
-  apt-get update \
+RUN apt-get update \
   && apt-get install -y \
     --no-install-recommends \
     build-essential=12.9 \
     gcc=4:10.2.* \
-    python3-dev=3.9.*
+    python3-dev>=${PYTHON_VERSION}
 
 # prepare pip for buildkit cache
+ARG APPNAME=InvokeAI
```

Comment:
already defined in previous stage

Reply:
Yes, but in podman the args get cleared out at every stage.  I don't know why but it took me longer than I care to admit to figure that out.

Reply:
But this is producing a "huge" maintanance overheap (one spot to change a default value vs three spots)
So if rly necesarry, then plz create env variables from the arguments in the base image and reuse those in the later stages, while cleaning them out to not polute the container

Reply:
ok I can try that.